developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,311 @@
+    return Buffer.from(arr);
+  }
+
+  static serialize_uint8(num: bigint): Buffer {
+    const arr = new ArrayBuffer(1);
+    const view = new DataView(arr);
+    view.setInt8(0, num);
+    return Buffer.from(arr);
+  }
+}
+
+function hashPrefix(): Buffer {
+  let hash = SHA3.sha3_256.create();
+  // APTOS::RawTransaction
+  hash.update(`${HASH_PREFIX}RawTransaction`);
+  return Buffer.from(hash.arrayBuffer());
+}
+
+function bcsSerializeTransactionPayload(payload: TransactionPayload): Buffer {
+  return Buffer.concat([
+    BCS.serialize_string(payload.type),
+    BCS.serialize_string(payload.function),
+    ...payload.type_arguments.map(tyArg => BCS.serialize_string(tyArg)),
+    ...payload.arguments.map(arg => BCS.serialize_string(arg))
+  ]);
+}
+
+function bcsSerializeRawTransaction(txn: RawTransaction): Buffer {
+  return Buffer.concat([
+    BCS.serialize_string(txn.sender),
+    BCS.serialize_uint64(txn.sequence_number),
+    bcsSerializeTransactionPayload(txn.payload),
+    BCS.serialize_uint64(txn.max_gas_amount),
+    BCS.serialize_uint64(txn.gas_unit_price),
+    BCS.serialize_uint64(txn.expiration_timestamp_secs),
+    BCS.serialize_uint8(txn.chain_id),
+  ]);
+}
+
+// This will hash a raw transaction into bytes
+function hashRawTransaction(txn: RawTransaction): Buffer {
+  // Generate a hash prefix
+  const prefix = hashPrefix();
+
+  // Serialize txn with BCS
+  const bcsSerializedTxn = bcsSerializeRawTransaction(txn);
+
+  return Buffer.concat([prefix, bcsSerializedTxn]);
+}
+
+```
+
+## Step 3) Sign the signing message with your account private key
+
+Sign the hashed raw transaction bytes with [Ed25519][ed25519]
+
+```typescript
+import * as Nacl from "tweetnacl";
+
+const rawTxn = createRawTransaction();
+const signature = Nacl.sign(hashRawTransaction(rawTxn), ACCOUNT_PRIVATE_KEY);
+```
+
+## Step 4) Create a SignedTransaction
+
+```typescript
+interface Authenticator {
+  public_key: Buffer,
+  signature: Buffer
+}
+
+interface SignedTransaction {
+  raw_txn: RawTransaction,
+  authenticator: Authenticator
+}
+
+const authenticator = {
+  public_key: PUBLIC_KEY,
+  signature: signature
+}
+
+const signedTransaction: SignedTransaction = {
+  raw_txn: rawTxn,
+  authenticator: authenticator
+};
+```
+
+
+## Step 5) Serialize SignedTransaction into bytes with BCS, and then hex-encode the bytes into a string.
+
+```typescript
+function bcsSerializeAuthenticator(authenticator: Authenticator): Buffer {
+  return Buffer.concat([
+    BCS.serialize_bytes(Uint8Array.from(authenticator.public_key)),
+    BCS.serialize_bytes(Uint8Array.from(authenticator.signature)),
+  ]);
+}
+
+function bcsSerializeSignedTransaction(txn: SignedTransaction): Buffer {
+  return Buffer.concat([
+    bcsSerializeRawTransaction(txn.raw_txn),
+    bcsSerializeAuthenticator(txn.authenticator)
+  ]);
+}
+
+const signedTransactionPayload = bcsSerializeSignedTransaction(signedTransaction).toString("hex");
+
+```
```

Comment:
it would be great to just show the layout and assume that the BCS serializer can translate that rather than showing all of the boiler plate code of serialize_bytes

Reply:
👍